lib/v2/screens/sign_up/claim_invite/components/bottom_container.dart

```diff
@@ -0,0 +1,178 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:seeds/i18n/claim_code.i18n.dart';
+import 'package:seeds/utils/string_extension.dart';
+import 'package:seeds/v2/components/flat_button_long.dart';
+import 'package:seeds/v2/components/quadstate_clipboard_icon_button.dart';
+import 'package:seeds/v2/constants/app_colors.dart';
+import 'package:seeds/v2/design/app_theme.dart';
+import 'package:seeds/v2/domain-shared/page_state.dart';
+import 'package:seeds/v2/navigation/navigation_service.dart';
+import 'package:seeds/v2/screens/sign_up/viewmodels/bloc.dart';
+import 'package:seeds/v2/screens/sign_up/viewmodels/states/claim_invite_state.dart';
+import 'package:seeds/v2/utils/debouncer.dart';
+import 'package:seeds/v2/utils/helpers.dart';
+
+class BottomContainer extends StatefulWidget {
+  const BottomContainer();
+
+  @override
+  _BottomContainerState createState() => _BottomContainerState();
+}
+
+class _BottomContainerState extends State<BottomContainer> {
+  late SignupBloc _signupBloc;
+  final _keyController = TextEditingController();
+  late ClaimInviteState _currentState;
```

Comment:
This is a blocker for me. 

We should not save states here or at all. 


I see you save this state so you can use it later  here `_inviteCodeSuffixIcon`

But this should be refactored into a flutter widget and in that widget you can use a `BlocBuilder` that will provide the latest state